src/core/thread/dua_manager.cpp

```diff
@@ -33,51 +33,106 @@
+    , mDuaState(kNotExist)
     , mDadCounter(0)
+    , mLastRegistrationTime(0)
+#endif
+#if OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+    , mChildIndexDuaRegistering(0)
+    , mRegisterCurrentChildIndex(false)
+#endif
 {
+    mDelay.mValue = 0;
+
+#if OPENTHREAD_CONFIG_DUA_ENABLE
     mDomainUnicastAddress.Clear();
     mDomainUnicastAddress.mAddressOrigin      = OT_ADDRESS_ORIGIN_THREAD;
-    mDomainUnicastAddress.mPreferred          = true;
+    mDomainUnicastAddress.mPreferred          = false;
     mDomainUnicastAddress.mValid              = true;
     mDomainUnicastAddress.mScopeOverride      = Ip6::Address::kGlobalScope;
     mDomainUnicastAddress.mScopeOverrideValid = true;
 
     mFixedDuaInterfaceIdentifier.Clear();
+#endif
+
+#if OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+    mChildDuaMask.ClearAll();
+    mChildDuaRegisteredMask.ClearAll();
+#endif
+
+    Get<Coap::Coap>().AddResource(mDuaNotification);
 }
 
-void DuaManager::UpdateDomainUnicastAddress(BackboneRouter::Leader::DomainPrefixState aState)
+void DuaManager::HandleDomainPrefixUpdate(BackboneRouter::Leader::DomainPrefixState aState)
 {
-    const otIp6Prefix *prefix;
+    const otIp6Prefix *prefix = nullptr;
+
+    if (aState == BackboneRouter::Leader::kDomainPrefixRefreshed)
+    {
+        if (mIsDuaPending)
+        {
+            IgnoreError(Get<Coap::Coap>().AbortTransaction(&DuaManager::HandleDuaResponse, this));
+        }
+
+#if OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+        if (mChildDuaMask.HasAny())
+        {
+            mChildDuaMask.ClearAll();
+            mChildDuaRegisteredMask.ClearAll();
+            mRegisterCurrentChildIndex = false;
+        }
+#endif
+    }
+
+    if ((aState == BackboneRouter::Leader::kDomainPrefixAdded) ||
+        (aState == BackboneRouter::Leader::kDomainPrefixRefreshed) ||
+        (aState == BackboneRouter::Leader::kDomainPrefixUnchanged))
+    {
+        prefix = Get<BackboneRouter::Leader>().GetDomainPrefix();
+
+        OT_ASSERT(prefix != nullptr);
+    }
 
+#if OPENTHREAD_CONFIG_DUA_ENABLE
     if ((aState == BackboneRouter::Leader::kDomainPrefixRemoved) ||
         (aState == BackboneRouter::Leader::kDomainPrefixRefreshed))
     {
-        Get<ThreadNetif>().RemoveUnicastAddress(mDomainUnicastAddress);
+        RemoveDomainUnicastAddress();
     }
 
-    VerifyOrExit((aState == BackboneRouter::Leader::kDomainPrefixAdded) ||
-                     (aState == BackboneRouter::Leader::kDomainPrefixRefreshed),
+    VerifyOrExit(aState == BackboneRouter::Leader::kDomainPrefixAdded ||
+                     aState == BackboneRouter::Leader::kDomainPrefixRefreshed ||
+                     (aState == BackboneRouter::Leader::kDomainPrefixUnchanged &&
+                      !Get<ThreadNetif>().HasUnicastAddress(
+                          GetDomainUnicastAddress())) /* in case removed for some reason e.g. the kDuaInvalid response
+                                                         from PBBR forcely */
+                 ,
```

Comment:
How about the following:
```suggestion
    switch (aState)
    {
    case BackboneRouter::Leader::kDomainPrefixAdded:
    case BackboneRouter::Leader::kDomainPrefixRefreshed:
        break;
    case BackboneRouter::Leader::kDomainPrefixUnchanged:
        VerifyOrExit(!Get<ThreadNetif>().HasUnicastAddress(GetDomainUnicastAddress()), OT_NOOP);
        break;
    default:
        ExitNow();
    }
```